Orange/widgets/evaluate/owcalibrationplot.py

```diff
@@ -117,106 +211,274 @@ def clear(self):
+                pen=pg.mkPen(color="k", style=Qt.DashLine, width=2),
+                hoverPen=pg.mkPen(color="k", style=Qt.DashLine, width=3),
+                bounds=(0, 1),
+            )
+            self.line.sigPositionChanged.connect(self.threshold_change)
+            self.line.sigPositionChangeFinished.connect(
+                self.threshold_change_done)
+            self.plot.addItem(self.line)
+
+    def _check_class_presence(self, ytrue):
+        self.Error.all_target_class.clear()
+        self.Error.no_target_class.clear()
+        if np.max(ytrue) == 0:
+            self.Error.no_target_class()
+            return False
+        if np.min(ytrue) == 1:
+            self.Error.all_target_class()
+            return False
+        return True
 
     def _replot(self):
         self.plot.clear()
         if self.results is not None:
             self._setup_plot()
+        self._update_info()
 
     def _on_display_rug_changed(self):
         self._replot()
 
+    def _on_selection_changed(self):
+        self._replot()
+        self.apply()
+
+    def threshold_change(self):
+        self.threshold = round(self.line.pos().x(), 2)
+        self.line.setPos(self.threshold)
+        self._update_info()
+
+    def _update_info(self):
+        text = f"""<table>
+                        <tr>
+                            <th align='right'>Threshold: p=</th>
+                            <td colspan='4'>{self.threshold:.2f}<br/></td>
+                        </tr>"""
+        if self.scores is not None:
+            short_names = Metrics[self.score].short_names
+            if short_names:
+                text += f"""<tr>
+                                <th></th>
+                                {"<td></td>".join(f"<td align='right'>{n}</td>"
+                                                  for n in short_names)}
+                            </tr>"""
+            for name, (probs, curves) in self.scores:
+                ind = min(np.searchsorted(probs, self.threshold),
+                          len(probs) - 1)
+                text += f"<tr><th align='right'>{name}:</th>"
+                text += "<td>/</td>".join(f'<td>{curve[ind]:.3f}</td>'
+                                          for curve in curves)
+                text += "</tr>"
+            text += "<table>"
+            self.info_label.setText(text)
+
+    def threshold_change_done(self):
+        self.apply()
+
+    def apply(self):
+        self.Information.no_output.clear()
+        wrapped = None
+        results = self.results
+        if results is not None:
+            problems = [
+                msg for condition, msg in (
+                    (len(results.folds) > 1,
+                     "each training data sample produces a different model"),
+                    (results.models is None,
+                     "test results do not contain stored models - try testing on"
+                     "separate data or on training data"),
```

Comment:
This produces a badly formatted warning, specifically "try testing onseparate data". Please introduce a space here or a newline.